gtda/time_series/embedding.py

```diff
@@ -208,52 +213,169 @@ def plot(Xt, sample=0, plotly_params=None):
         return plot_point_cloud(Xt[sample], plotly_params=plotly_params)
 
 
+def time_delay_embedding(
+        X, time_delay=1, dimension=2, stride=1, flatten=False,
+        ensure_last_value=True, validate=True
+        ):
+    """Time-delay embeddings of arrays of time-series data.
+
+    On a 1D array `X` representing a single time series, the time-delay
+    embedding algorithm is the one described in :class:`TakensEmbedding` and
+    yields a 2D array representing a point cloud in Euclidean space. 2D array
+    inputs are interpreted as collections of time series (one per entry along
+    axis 0), and the algorithm is applied to each time series separately to
+    produce a 3D array of point clouds. More generally, N-dimensional array
```

Comment:
Sorry for the confusion, but I am not sure I understand this data structure. If there is n_time_series of n_stamps, shouldn't the shape be (n_time_series, n_staps)?

Reply:
The shape indeed is (n_time_series, n_timestamps) for 2D arrays -- see the information on `X` in "Parameters". There is no contradiction with what's written here, as here I am saying (N=2) "there there is one time series per entry along the first N-1=1 axes", i.e. one time series per entry along axis 0.